### PR TITLE
Revamp: Grid Filtering

### PR DESCRIPTION
**Pull request by svdimitr**

### CODE REVIEW

File: components/grid/filter/overview.md [hunk 1]
```diff
@@ -9,43 +9,43 @@ previous_url: /components/grid/filtering
...
+## More Filtering Options
 
-You can prevent the user from filtering a certain field by setting `Filterable="false"` on its column.
+In addition to the two main filtering modes, the grid offers two more features that can enhance the user experience when looking for data.
 
-## More Filtering Options
+### ToolBar SearchBox
 
-In addition to the two main filtering modes, the grid offers two more features that can enhance the user experience when looking for data:
+The ToolBar of the Telerik Grid for Blazor includes a [SearchBox]({%slug grid-searchbox%}), that lets users amend filters and search across multiple fields simultaneously.
```
Comment (yordan-mitev) [inline components/grid/filter/overview.md:32]:
```suggestion
The ToolBar of the Telerik Grid for Blazor includes a [SearchBox]({%slug grid-searchbox%}) that lets users amend filters and search across multiple fields simultaneously.
```

File: components/grid/filter/overview.md [hunk 2]
```diff
@@ -9,43 +9,43 @@ previous_url: /components/grid/filtering
...
-* A [searchbox in the toolbar]({%slug grid-searchbox%}) can amend the filters and let the user look up many fields at once
+### CheckBoxList
 
-* The filter menu can show a [list of checkboxes]({%slug grid-checklist-filter%}) with the distinct values from the data to make filtering resemble Excel.
+The filter menu can display a [list of checkboxes]({%slug grid-checklist-filter%}) with distinct values from the data, making filtering similar to Excel.
 
 ## Filter Descriptors
 
-The Grid filter state is stored in [CompositeFilterDescriptors](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor). The below information is important if you want to [get or change the Grid filters programmatically]({%slug grid-state%}).
+The Grid filter state is stored in [CompositeFilterDescriptors](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor). Use the following information if you want to [get or change the Grid filters programmatically]({%slug grid-state%}).
```
Comment (yordan-mitev) [inline components/grid/filter/overview.md:40]:
```suggestion
The Grid filter state is stored in [`CompositeFilterDescriptors`](/blazor-ui/api/Telerik.DataSource.CompositeFilterDescriptor). Use the following information if you want to [get or change the Grid filters programmatically]({%slug grid-state%}).
```

File: components/grid/filter/overview.md [hunk 3]
```diff
@@ -9,43 +9,43 @@ previous_url: /components/grid/filtering
...
 
-Each `CompositeFilterDescriptor` contains a [**collection** of `FilterDescriptor`s](/blazor-ui/api/Telerik.DataSource.FilterDescriptorCollection). All descriptors in the collection are applied with an *AND* or an *OR* `LogicalOperator`.
+Each CompositeFilterDescriptor includes a collection of FilterDescriptors, where all descriptors in the collection are applied with an AND or OR LogicalOperator.
```
Comment (yordan-mitev) [inline components/grid/filter/overview.md:42]: Don't invent new words when referring to well-known concepts&mdash;follow the common writing standards.
```suggestion
Each `CompositeFilterDescriptor` includes a collection of filter descriptors, where all descriptors in the collection are applied with an AND or OR logical operator.
```

File: components/grid/filter/overview.md [hunk 4]
```diff
@@ -9,43 +9,43 @@ previous_url: /components/grid/filtering
...
 
-* [Filter Row]({%slug grid-filter-row%}) - each `CompositeFilterDescriptor` targets a specific field. By default, one filter can be applied to a field using the Filter Row operator. The filter value is stored in the first `FilterDescriptor` instance of the `CompositeFilterDescriptor` for that field.
+* [Filter Row]({%slug grid-filter-row%}) - Each `CompositeFilterDescriptor` targets a specific field. By default, one filter can be applied to a field using the Filter Row operator, with the filter value stored in the first FilterDescriptor instance for that field.
```
Comment (yordan-mitev) [inline components/grid/filter/overview.md:44]: Use an em dash. The same comment applies to the next two lines.
```suggestion
* [Filter Row]({%slug grid-filter-row%})&mdash;Each `CompositeFilterDescriptor` targets a specific field. By default, one filter can be applied to a field using the Filter Row operator, with the filter value stored in the first filter descriptor instance for that field.
```

File: components/grid/filter/overview.md [hunk 5]
```diff
@@ -9,43 +9,43 @@ previous_url: /components/grid/filtering
...
 
-* [Filter Menu]({%slug grid-filter-menu%}) - each `CompositeFilterDescriptor` targets a specific field. Filter values from the separate filter opearators in the menu are stored in different `FilterDescriptor` instances of the dedicated `CompositeFilterDescriptor` for that field.
+* [Filter Menu]({%slug grid-filter-menu%}) - Each `CompositeFilterDescriptor` targets a specific field. The filter values from separate filter operators in the menu are stored in different FilterDescriptor instances within the CompositeFilterDescriptor for that field.
```
Comment (yordan-mitev) [inline components/grid/filter/overview.md:46]:
```suggestion
* [Filter Menu]({%slug grid-filter-menu%})&mdash;Each `CompositeFilterDescriptor` targets a specific field. The filter values from separate filter operators in the menu are stored in different filter descriptor instances within the `CompositeFilterDescriptor` for that field.
```

File: components/grid/filter/overview.md [hunk 6]
```diff
@@ -9,43 +9,43 @@ previous_url: /components/grid/filtering
...
 
-* [SearchBox]({%slug grid-searchbox%}) - one `CompositeFilterDescriptor` is created in the state when the user types in the Searchbox. By default, it targets all `string` fields. A dedicated `FilterDescriptor` instance is added to this `CompositeFilterDescriptor` for each `string` field. Each `FilterDescriptor` instance contains the filter value typed in the Searchbox.
+* [SearchBox]({%slug grid-searchbox%}) - A `CompositeFilterDescriptor` is created in the state when the user types in the Searchbox. By default, it targets all string fields, adding a dedicated FilterDescriptor instance for each string field. Each FilterDescriptor contains the filter value typed in the SearchBox.
```
Comment (yordan-mitev) [inline components/grid/filter/overview.md:48]:
```suggestion
* [SearchBox]({%slug grid-searchbox%})&mdash;A `CompositeFilterDescriptor` is created in the state when the user types in the search box. By default, it targets all string fields, adding a dedicated filter descriptor instance for each string field. Each filter descriptor contains the filter value typed in the search box.
```

File: components/grid/filter/overview.md [hunk 7]
```diff
@@ -132,7 +132,7 @@ The following articles and sample projects can be helpful when implementing filt
 
 ## Notes
 
-* The grid uses `Activator.CreateInstance<TItem>();` to get the type of the item it is bound to in order to generate proper filters and filter operators for them. Thus, the Model should have a Parameterless constructor defined.
+The grid uses `Activator.CreateInstance<TItem>();` to obtain the item's type it is bound to, enabling it to generate accurate filters and filter operators. To facilitate this, ensure that the Model has a parameterless constructor defined. If your model has no parameterless constructor use the [OnModelInit event]({%slug grid-events%}#onmodelinit).
```
Comment (yordan-mitev) [inline components/grid/filter/overview.md:135]: Capital "G"
```suggestion
The Grid uses `Activator.CreateInstance<TItem>();` to obtain the item's type it is bound to, enabling it to generate accurate filters and filter operators. To facilitate this, ensure you define a parameterless constructor for the model. If your model has no parameterless constructor use the [`OnModelInit` event]({%slug grid-events%}#onmodelinit).
```

File: components/grid/filter/filter-row.md [hunk 8]
```diff
@@ -75,23 +68,13 @@ The default filter operator is `Contains` for `string` columns and `IsEqualTo` f
 
 ## Filter From Code
 
-You can set the grid filters from your code through the grid [state]({%slug grid-state%}).
+To programatically filter the Grid, read the [Grid State]({%slug grid-state%}) documentation article.
```
[the remark this answers is not in the document]
Comment (svdimitr) [reply components/grid/filter/filter-row.md:71]: I saw the comment, however, I am unsure why it is better, the sentence is in an active voice already and is grammatically correct. The "To learn how to..." is a new construct that we have not used before, on many occasions, we used "read the X article" which I used here as well.

**Review by yordan-mitev — APPROVED** [review]

**Pull request merged**